### PR TITLE
refator: setting page

### PR DESCRIPTION
**Pull request by alantoa**
# Why
### Settings page has some issues:
- [sometimes crash](https://linear.app/showtime/issue/SHOW2-966/crash-when-switching-settings-tabs)
- sometimes scroll it back to the top.
- desktop still uses the mobile's UI, however, many users currently use the desktop, and we need to improve it!
- when adding an email, its header will be obscured.

# How

- refactor the Setting using the new `HeaderTabView`, and adjust the UI to make the desktop look beautiful.
- move `addEmail` Modal to ModalScreen.

# Test Plan

check if the Settings page looks good!

### CODE REVIEW

File: packages/app/components/swipe-list.tsx [hunk 1]
```diff
@@ -358,7 +358,7 @@ export const FeedItem = memo(
                 </Suspense>
               </View>
               <Description nft={nft} />
-              <View tw="item-center flex-row justify-between px-4">
+              <View tw="item-centers flex-row justify-between px-4">
```
Comment (axeldelafosse) [inline packages/app/components/swipe-list.tsx]: `items-center`

File: packages/app/navigation/linking.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ const linking: LinkingOptions<ReactNavigation.RootParamList> = {
       followers: "profile/followers",
       following: "profile/following",
       settings: "settings",
+      addEmail: "settings/addEmail",
```
Comment (axeldelafosse) [inline packages/app/navigation/linking.ts]: Nit for consistency: `settings/add-email`

**Review by axeldelafosse — APPROVED** [review]
LFG ✨

File: packages/app/components/swipe-list.tsx (hunk 1 above)
Comment (alantoa) [reply packages/app/components/swipe-list.tsx]: damn, the pointer is wrong.🤣

**Review by intergalacticspacehighway — APPROVED** [review]
LGTM 🔥

**Pull request merged**